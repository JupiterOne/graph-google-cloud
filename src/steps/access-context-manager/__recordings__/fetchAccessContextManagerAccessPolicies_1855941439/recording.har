{
  "log": {
    "_recordingName": "fetchAccessContextManagerAccessPolicies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 572,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 572,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:23 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.123Z",
        "time": 212,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 212
        }
      },
      {
        "_id": "e7ae7a9d3285d95d4330a7c239aa61b7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.15.4 auth/7.1.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "accesscontextmanager.googleapis.com"
            }
          ],
          "headersSize": 581,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "parent",
              "value": "organizations/958457776463"
            }
          ],
          "url": "https://accesscontextmanager.googleapis.com/v1/accessPolicies?parent=organizations%2F958457776463"
        },
        "response": {
          "bodySize": 369,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 369,
            "text": "{\"accessPolicies\":[{\"name\":\"accessPolicies/368605946207\",\"parent\":\"organizations/958457776463\",\"title\":\"default policy\",\"etag\":\"64186ca0c82b01b4\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:24 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.374Z",
        "time": 829,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 829
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
