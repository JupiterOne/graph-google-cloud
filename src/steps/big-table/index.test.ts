import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import {
  fetchAppProfiles,
  fetchBackups,
  fetchClusters,
  fetchInstances,
  fetchTables,
} from '.';
import { integrationConfig } from '../../../test/config';
import { setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import {
  ENTITY_TYPE_BIG_TABLE_APP_PROFILE,
  ENTITY_TYPE_BIG_TABLE_BACKUP,
  ENTITY_TYPE_BIG_TABLE_CLUSTER,
  ENTITY_TYPE_BIG_TABLE_INSTANCE,
  ENTITY_TYPE_BIG_TABLE_TABLE,
  RELATIONSHIP_TYPE_CLUSTER_HAS_BACKUP,
  RELATIONSHIP_TYPE_CLUSTER_USES_KMS_KEY,
  RELATIONSHIP_TYPE_INSTANCE_HAS_APP_PROFILE,
  RELATIONSHIP_TYPE_INSTANCE_HAS_CLUSTER,
  RELATIONSHIP_TYPE_INSTANCE_HAS_TABLE,
  RELATIONSHIP_TYPE_TABLE_HAS_BACKUP,
} from './constants';

const tempNewAccountConfig = {
  ...integrationConfig,
  serviceAccountKeyFile: integrationConfig.serviceAccountKeyFile.replace(
    'j1-gc-integration-dev-v2',
    'j1-gc-integration-dev-v3',
  ),
  serviceAccountKeyConfig: {
    ...integrationConfig.serviceAccountKeyConfig,
    project_id: 'j1-gc-integration-dev-v3',
  },
};

describe('#fetchInstances', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchInstances',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: tempNewAccountConfig,
    });

    await fetchInstances(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === ENTITY_TYPE_BIG_TABLE_INSTANCE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Database'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_instance' },
          name: { type: 'string' },
          state: { type: 'string' },
          type: { type: 'string' },
          webLink: { type: 'string' },
        },
      },
    });
  });
});

describe('#fetchAppProfiles', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchAppProfiles',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: tempNewAccountConfig,
    });

    await fetchInstances(context);
    await fetchAppProfiles(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === ENTITY_TYPE_BIG_TABLE_APP_PROFILE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Configuration'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_app_profile' },
          name: { type: 'string' },
          instanceId: { type: 'string' },
          etag: { type: 'string' },
          description: { type: 'string' },
          webLink: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_INSTANCE_HAS_APP_PROFILE,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_instance_has_app_profile',
          },
        },
      },
    });
  });
});

describe('#fetchClusters', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchClusters',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: tempNewAccountConfig,
    });

    await fetchInstances(context);
    await fetchClusters(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === ENTITY_TYPE_BIG_TABLE_CLUSTER,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Cluster'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_cluster' },
          name: { type: 'string' },
          instanceId: { type: 'string' },
          state: { type: 'string' },
          location: { type: 'string' },
          defaultStorageType: { type: 'string' },
          serveNodes: { type: 'number' },
          kmsKeyName: { type: 'string' },
          webLink: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_INSTANCE_HAS_CLUSTER,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_instance_has_cluster',
          },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_CLUSTER_USES_KMS_KEY,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'USES' },
          _type: {
            const: 'google_bigtable_cluster_uses_kms_key',
          },
        },
      },
    });
  });
});

describe('#fetchBackups', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchBackups',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: tempNewAccountConfig,
    });

    await fetchInstances(context);
    await fetchClusters(context);
    await fetchBackups(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === ENTITY_TYPE_BIG_TABLE_BACKUP,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Backup'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_backup' },
          name: { type: 'string' },
          instanceId: { type: 'string' },
          clusterId: { type: 'string' },
          sourceTable: { type: 'string' },
          expireTime: { type: 'string' },
          startTime: { type: 'string' },
          endTime: { type: 'string' },
          sizeBytes: { type: 'string' },
          state: { type: 'string' },
          encryptionType: { type: 'string' },
          kmsKeyVersion: { type: 'string' },
          webLink: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_CLUSTER_HAS_BACKUP,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_cluster_has_backup',
          },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_TABLE_HAS_BACKUP,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_table_has_backup',
          },
        },
      },
    });
  });
});

describe('#fetchTables', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchTables',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: tempNewAccountConfig,
    });

    await fetchInstances(context);
    await fetchTables(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === ENTITY_TYPE_BIG_TABLE_TABLE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['DataCollection'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigtable_table' },
          name: { type: 'string' },
          instanceId: { type: 'string' },
          granularity: { type: 'string' },
          backup: { type: 'string' },
          webLink: { type: 'string' },
          classification: { const: 'null' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_INSTANCE_HAS_TABLE,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigtable_instance_has_table',
          },
        },
      },
    });
  });
});
