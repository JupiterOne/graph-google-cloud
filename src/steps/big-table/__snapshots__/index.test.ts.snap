// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchAppProfiles should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Database",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    {
      "_class": [
        "Configuration",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
            "singleClusterRouting": {
              "allowTransactionalWrites": true,
              "clusterId": "bt-instance",
            },
          },
        },
      ],
      "_type": "google_bigtable_app_profile",
      "createdOn": undefined,
      "description": undefined,
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "etag": undefined,
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "singleClusterRouting.allowTransactionalWrites": true,
      "singleClusterRouting.clusterId": "bt-instance",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/app-profiles/bt-profile?project=j1-gc-integration-dev-v3",
    },
    {
      "_class": [
        "Configuration",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "description": "Default application profile for this instance. This profile is used if you do not supply a different profile ID at connection time.",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
            "singleClusterRouting": {
              "allowTransactionalWrites": true,
              "clusterId": "bt-instance",
            },
          },
        },
      ],
      "_type": "google_bigtable_app_profile",
      "createdOn": undefined,
      "description": "Default application profile for this instance. This profile is used if you do not supply a different profile ID at connection time.",
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "etag": undefined,
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "singleClusterRouting.allowTransactionalWrites": true,
      "singleClusterRouting.clusterId": "bt-instance",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/app-profiles/default?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": [
    {
      "_class": "HAS",
      "_fromEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance|has|projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "_toEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/bt-profile",
      "_type": "google_bigtable_instance_has_app_profile",
      "displayName": "HAS",
    },
    {
      "_class": "HAS",
      "_fromEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance|has|projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "_toEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance/appProfiles/default",
      "_type": "google_bigtable_instance_has_app_profile",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": [
    "google_bigtable_instance",
    "google_bigtable_app_profile",
    "google_bigtable_instance_has_app_profile",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 2,
}
`;

exports[`#fetchBackups should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Database",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    {
      "_class": [
        "Cluster",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "defaultStorageType": "HDD",
            "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
            "serveNodes": 3,
            "state": "READY",
          },
        },
      ],
      "_type": "google_bigtable_cluster",
      "active": true,
      "createdOn": undefined,
      "defaultStorageType": "HDD",
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "kmsKeyName": undefined,
      "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "serveNodes": 3,
      "state": "READY",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    {
      "_class": [
        "Backup",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "encryptionInfo": {
              "encryptionType": "GOOGLE_DEFAULT_ENCRYPTION",
            },
            "endTime": "2021-08-12T00:46:40.026785Z",
            "expireTime": "2021-08-13T00:46:37.503Z",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
            "sourceTable": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
            "startTime": "2021-08-12T00:46:39.326777Z",
            "state": "READY",
          },
        },
      ],
      "_type": "google_bigtable_backup",
      "active": true,
      "clusterId": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "createdOn": undefined,
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "encryptionType": "GOOGLE_DEFAULT_ENCRYPTION",
      "endTime": 1628729200026,
      "expireTime": 1628815597503,
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "kmsKeyVersion": undefined,
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "sizeBytes": undefined,
      "sourceTable": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "startTime": 1628729199326,
      "state": "READY",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/backups?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": [
    {
      "_class": "HAS",
      "_fromEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance|has|projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_toEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_type": "google_bigtable_instance_has_cluster",
      "displayName": "HAS",
    },
    {
      "_class": "HAS",
      "_fromEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance|has|projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "_toEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance/backups/tf-backup",
      "_type": "google_bigtable_cluster_has_backup",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": [
    "google_bigtable_instance",
    "google_bigtable_cluster",
    "google_bigtable_instance_has_cluster",
    "google_bigtable_backup",
    "google_bigtable_cluster_has_backup",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 2,
}
`;

exports[`#fetchClusters should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Database",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    {
      "_class": [
        "Cluster",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "defaultStorageType": "HDD",
            "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
            "serveNodes": 3,
            "state": "READY",
          },
        },
      ],
      "_type": "google_bigtable_cluster",
      "active": true,
      "createdOn": undefined,
      "defaultStorageType": "HDD",
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "kmsKeyName": undefined,
      "location": "projects/j1-gc-integration-dev-v3/locations/us-central1-b",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "serveNodes": 3,
      "state": "READY",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": [
    {
      "_class": "HAS",
      "_fromEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance|has|projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_toEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance/clusters/bt-instance",
      "_type": "google_bigtable_instance_has_cluster",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": [
    "google_bigtable_instance",
    "google_bigtable_cluster",
    "google_bigtable_instance_has_cluster",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;

exports[`#fetchInstances should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Database",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": [],
  "encounteredTypes": [
    "google_bigtable_instance",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchTables should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Database",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "displayName": "bt-instance",
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
            "state": "READY",
            "type": "PRODUCTION",
          },
        },
      ],
      "_type": "google_bigtable_instance",
      "active": true,
      "createdOn": undefined,
      "displayName": "bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "state": "READY",
      "type": "PRODUCTION",
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/overview?project=j1-gc-integration-dev-v3",
    },
    {
      "_class": [
        "DataCollection",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
          },
        },
      ],
      "_type": "google_bigtable_table",
      "backup": undefined,
      "classification": null,
      "createdOn": undefined,
      "displayName": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "granularity": undefined,
      "instanceId": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "name": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "sourceType": undefined,
      "webLink": "https://console.cloud.google.com/bigtable/instances/bt-instance/tables?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": [
    {
      "_class": "HAS",
      "_fromEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance",
      "_key": "projects/j1-gc-integration-dev-v3/instances/bt-instance|has|projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "_toEntityKey": "projects/j1-gc-integration-dev-v3/instances/bt-instance/tables/tf-table",
      "_type": "google_bigtable_instance_has_table",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": [
    "google_bigtable_instance",
    "google_bigtable_table",
    "google_bigtable_instance_has_table",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;
